### DISCUSSION TITLE
v.0.4 introduces a breaking change on how to pass values to attributes

### DISCUSSION THREAD

**Discussion by jpsca** [jpsca/jinjax#78] [General]
Since version 0.4, to pass values to components as attributes, do not use `name={value}` anymore. Instead:

1. For strings, keep using regular quotes:  `name="string"`
2. For values, use regular quotes but **prefix the name with a colon**: `:name="value"`.

For example, before:

```html+jinja
<Example columns={2} tabbed={False} panels={{'one': 'lorem', 'two': 'ipsum'}} />
``` 

is now:

```html+jinja
<Example :columns="2" :tabbed="False" :panels="{'one': 'lorem', 'two': 'ipsum'}" />
``` 

### How to update:

![image](https://github.com/jpsca/jinjax/assets/67524204/5120ce03-1f88-42f5-8298-0d9ebcd94bcd)

1. Open the search-and-replace of your text editor (VisualStudio Code in this image).
3. Activate using regular expressions
4. Toggle "replace"
5. Search for `(\w+)=\{(.*?)\}`
    Replace for `:$1="$2"`
    Only in `*.jinja` files

**Comment by pietz** [comment]
I'm not convinced this is a good change. I don't see any benefits over the old syntax. If anything it's slightly worse, I think. Before, you had to change the value portion in order to switch from a fixed value to a variable. Now you have to change the attribute name portion, which makes slightly less sense.

Is there a reason why JinjaX doesn't just use Jinja style variables with double curly braces?

```
<Example columns={{ 2 }} tabbed={{ False }} panels={{ {'one': 'lorem', 'two': 'ipsum'} }} />
```

**Reply by jpsca** [reply]
The double curly braces were one of the possibilities, one of the reasons to not use them is to make it clear that they are not actually Jinja variables.

For example, this is valid: `<div class="level-{{ indent }} bg-{{ color }}-500"></div>`
But this is not: ` <Card class="level-{{ indent }} bg-{{ color }}-500"></Card>`

However, I have to agree that in your example they actually look good. The single curly brace I think it was a mistake, but maybe JinjaX could support the new (vue-like) syntax **and** the double curly braces...

**Reply by jpsca** [reply]
```html+jinja
<FamousLastWords>
    There is no way that supporting both styles will be a problem in the future.
<FamousLastWords>
```

@pietz: Here we go

https://github.com/jpsca/jinjax/pull/80
https://github.com/jpsca/jinjax/releases/tag/0.42

**Reply by pietz** [reply]
You rock!

**Comment by pietz** [comment]
Highly subjective, I get it, but the double curly braces I like the most by far. Just updated my app.
